core/services/keystore/keys/ocr2key/terra_keyring.go

```diff
@@ -84,12 +71,12 @@ func (ok *terraKeyring) MaxSignatureLength() int {
 }
 
 func (ok *terraKeyring) marshal() ([]byte, error) {
-	return ok.secret, nil
+	return ok.privKey.Seed(), nil
 }
 
 func (ok *terraKeyring) unmarshal(in []byte) error {
-	key := cosmosed25519.GenPrivKeyFromSecret(in)
-	ok.PrivKey = key
-	ok.secret = in
+	privKey := ed25519.NewKeyFromSeed(in)
+	ok.privKey = privKey
+	ok.pubKey = ed25519.PublicKey(privKey[32:])
```

Comment:
seems safer to just use privKey.Public() ?

Reply:
that returns an interface though, and it feels a bit clunky to extract the value from the interface